### PR TITLE
Fuel merge part 10 - Fuel specific names for transaction fields

### PR DESCRIPTION
**Pull request by DZakh**

### CODE REVIEW

File: codegenerator/cli/templates/static/codegen/src/EventFetching.res [hunk 1]
```diff
@@ -88,13 +88,16 @@ let getTxFieldFromEthersLog = (log: Ethers.log, txField: string, ~logger): txFie
...
   //Note: if we implement all transaction fields, we will need all
   //field names not just non optional ones
-  Types.Transaction.nonOptionalFieldNames
-  ->Belt.Array.map(name => (name, getTxFieldFromEthersLog(log, name, ~logger)))
-  ->Js.Dict.fromArray
-  ->(Utils.magic: Js.Dict.t<txFieldVal> => Types.Transaction.t)
+  nonOptionalTransactionFieldNames->Belt.Array.forEach(name => {
+    dict->Js.Dict.set(name, getTxFieldFromEthersLog(log, name, ~logger))
+  })
+  dict->(Utils.magic: Js.Dict.t<txFieldVal> => Types.Transaction.t)
```
Comment (DZakh) [inline codegenerator/cli/templates/static/codegen/src/EventFetching.res:100]: Optimised it a little bit

File: codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res [hunk 2]
```diff
@@ -3,112 +3,73 @@ open Belt
...
-    pageFetchTime: int,
-  }
-
-  let waitForBlockGreaterThanCurrentHeight = (~currentBlockHeight, ~logger) => {
-    HyperSync.pollForHeightGtOrEq(
-      ~serverUrl=T.endpointUrl,
-      ~blockNumber=currentBlockHeight,
-      ~logger,
-    )
+let makeGetNextPage = (~endpointUrl, ~contracts: array<Config.contract>, ~queryLogsPage, ~pollForHeightGtOrEq, ~blockSchema, ~transactionSchema) => {
```
Comment (DZakh) [inline codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res:53]: Moved it outiside of the functor to simplify testing

File: codegenerator/cli/templates/static/codegen/src/eventFetching/hypersync/HyperSync.res [hunk 3]
```diff
@@ -92,46 +92,39 @@ module CachedClients = {
...
-
+  let convertEvent = (event: HyperSyncClient.ResponseTypes.event, ~nonOptionalBlockFieldNames, ~nonOptionalTransactionFieldNames): logsQueryPageItem => {
+    let missingParams = []
+    missingParams->addMissingParams(Types.Log.fieldNames, event.log, ~prefix="log")
+    missingParams->addMissingParams(nonOptionalBlockFieldNames, event.block, ~prefix="block")
+    missingParams->addMissingParams(
+      nonOptionalTransactionFieldNames,
+      event.transaction,
+      ~prefix="transaction",
+    )
```
Comment (DZakh) [inline codegenerator/cli/templates/static/codegen/src/eventFetching/hypersync/HyperSync.res:127]: Still the same validation, but should be a little bit more performant

File: scenarios/test_codegen/test/HyperSyncWorker_test.res [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    toBlock: int,
+    logSelections: array<LogSelection.t>,
+    fieldSelection: HyperSyncClient.QueryTypes.fieldSelection,
+    nonOptionalBlockFieldNames: array<string>,
+    nonOptionalTransactionFieldNames: array<string>,
+  }
+
+  type mock = {queryLogsPageCalls: array<queryLogsPageCall>}
+
+  let mock = (~blockSchema=S.object(_ => ()), ~transactionSchema=S.object(_ => ())) => {
```
Comment (DZakh) [inline scenarios/test_codegen/test/HyperSyncWorker_test.res:16]: This mock is not the most flexible, but can be extended to conveniently test the logSelections logic, as well as polling logic

**Review by JonoPrest — APPROVED** [review]
Awesome Dmitry looks great 👌🏼 🙏🏼

**Pull request merged**